src/nested_pandas/nestedframe/core.py

```diff
@@ -258,7 +258,17 @@ def _getitem_list(self, item):
         return result
 
     def __setitem__(self, key, value):
-        """Adds custom __setitem__ behavior for nested columns"""
+        """Custom __setitem__ for NestedFrame: auto-nest DataFrame assignment to new columns."""
+        # If assigning a DataFrame to a new column, auto-nest it
+        if (
+            isinstance(key, str)
+            and key not in self.columns
+            and isinstance(value, (pd.DataFrame | NestedFrame))
+        ):
+            new_df = self.add_nested(value, name=key)
```

Comment:
Could you add a comment here saying that because this is add_nested, this is essentially a left join? Just wanting to note that this will affect the resulting data in the matched nested dataframe